url24d793285d_auzll_gengweibo_tgz-pJ8-org_json_TestJ8/src/com/gengweibo/dao/mysql/WeiboDaoJdbcImpl.java

```diff
@@ -0,0 +1,275 @@
+/**
+ * http://auzll.iteye.com/
+ */
+package com.gengweibo.dao.mysql;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.Collections;
+import java.util.List;
+
+import javax.sql.DataSource;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import com.gengweibo.dao.WeiboDao;
+import com.gengweibo.weibo.IWeibo;
+import com.gengweibo.weibo.Weibo;
+import com.gengweibo.weibo.WeiboManager;
+import com.gengweibo.weibo.WeiboType;
+import com.google.inject.Inject;
+import com.google.inject.internal.Lists;
+
+/**
+ * @author auzll@msn.com
+ * @since 2011-04-06
+ */
+public class WeiboDaoJdbcImpl implements WeiboDao {
+    private static final Log LOG = LogFactory.getLog(WeiboDaoJdbcImpl.class);
+
+    @Inject
+    private DataSource dataSource;
+
+    private void closeQuietly(Connection conn) {
+        if (null != conn) {
+            try {
+                conn.close();
+            } catch (SQLException e) {
+                if (LOG.isDebugEnabled()) {
+                    LOG.debug("method:closeQuietly", e);
+                }
+            }
+        }
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 38:46, 50:58, 62:70.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.